moveit_py/moveit/utils.py

```diff
@@ -0,0 +1,58 @@
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2022, Peter David Fagan
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above
+#    copyright notice, this list of conditions and the following
+#    disclaimer in the documentation and/or other materials provided
+#    with the distribution.
+#  * Neither the name of Willow Garage, Inc. nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Author: Peter David Fagan
+
+import sys
+import traceback
+import yaml
+
+from tempfile import NamedTemporaryFile
+
+
+def create_params_file_from_dict(params, node_name):
+    with NamedTemporaryFile(mode="w", prefix="launch_params_", delete=False) as h:
+        param_file_path = h.name
+        param_dict = {node_name: {"ros__parameters": params}}
+        yaml.dump(param_dict, h, default_flow_style=False)
+        return param_file_path
+
+
+def get_launch_params_filepath():
+    """
+    A utility that returns the path value after argument --param-file.
```

Comment:
```suggestion
    A utility that returns the path value after argument --params-file.
```

Also more of a general question, but is there an opportunity to use the `argparse` module instead of this? Or is there something specific to how this work was done to prevent it?

Reply:
It has been a while since I wrote this, I believe at the time there was justification for not using argparse (this is what I normally use), but I need to revisit this to check. I am going to update the docstring but keep this item open for now.